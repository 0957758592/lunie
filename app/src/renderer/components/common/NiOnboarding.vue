<template lang="pug">
.ni-session-wrapper
  img.ni-session-backdrop(src="~assets/images/cosmos-logo.png")
  onboarding-welcome(v-if="onboarding.state == 'welcome'")
  onboarding-send(v-if="onboarding.state == 'send'")
  onboarding-stake(v-if="onboarding.state == 'stake'")
  onboarding-vote(v-if="onboarding.state == 'vote'")
  onboarding-explore(v-if="onboarding.state == 'explore'")
</template>

<script>
import { mapGetters } from "vuex"
import OnboardingWelcome from "common/NiOnboardingWelcome"
import OnboardingSend from "common/NiOnboardingSend"
import OnboardingStake from "common/NiOnboardingStake"
import OnboardingVote from "common/NiOnboardingVote"
import OnboardingExplore from "common/NiOnboardingExplore"
export default {
  name: "ni-onboarding",
  components: {
    OnboardingWelcome,
    OnboardingSend,
    OnboardingStake,
    OnboardingVote,
    OnboardingExplore
  },
  computed: { ...mapGetters(["onboarding"]) }
}
</script>
