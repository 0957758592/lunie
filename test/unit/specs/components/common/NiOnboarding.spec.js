import NiOnboarding from "common/NiOnboarding.vue"
import htmlBeautify from "html-beautify"
import setup from "../../../helpers/vuex-setup"

describe("NiOnboarding", () => {
  let wrapper, store, instance
  let { mount } = setup()

  beforeEach(() => {
    instance = mount(NiOnboarding, {
      stubs: { "app-menu": "<app-menu />" },
      methods: { watchWindowSize: () => {} }
    })
    wrapper = instance.wrapper
    store = instance.store

    wrapper.update()
  })

  it("has the expected html structure 1", () => {
    store.commit("setOnboardingActive", true)
    store.commit("setOnboardingState", "0")
    wrapper.update()
    expect(htmlBeautify(wrapper.html())).toMatchSnapshot()
  })
})
