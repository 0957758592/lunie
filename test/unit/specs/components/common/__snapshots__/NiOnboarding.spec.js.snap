// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NiOnboarding has the expected html structure 1 1`] = `
"<div class=\\"ni-session-wrapper\\">
   <img src=\\"~assets/images/cosmos-logo.png\\" class=\\"ni-session-backdrop\\">
   <div class=\\"ni-session\\">
      <div class=\\"ni-session-container\\">
         <div class=\\"ni-session-header\\">
            <div class=\\"ni-session-title\\">
               Welcome to Voyager
            </div>
</div>
<div class=\\"ni-session-main\\">
   <div class=\\"ni-session-label\\"></div>
<img>
<div class=\\"ni-bar-discrete\\">
   <div class=\\"ni-bar-discrete__container\\">
      <div class=\\"ni-bar-discrete__edge\\"></div>
<div class=\\"ni-bar-discrete__node\\"></div>
<div class=\\"ni-bar-discrete__node\\"></div>
<div class=\\"ni-bar-discrete__node\\"></div>
<div class=\\"ni-bar-discrete__node\\"></div>
<div class=\\"ni-bar-discrete__node\\"></div>
</div>
</div>
</div>
<div class=\\"ni-session-footer\\">
   <button class=\\"ni-btn\\"><span class=\\"ni-btn__container\\"><i aria-hidden=\\"true\\" class=\\"ni-btn__icon material-icons\\">close</i>
   <!---->
   <span class=\\"ni-btn__value\\">Skip</span></span></button><button class=\\"ni-btn\\"><span class=\\"ni-btn__container ni-btn__icon-right ni-btn--primary\\"><i aria-hidden=\\"true\\" class=\\"ni-btn__icon material-icons\\">chevron_right</i>
   <!---->
   <span class=\\"ni-btn__value\\">Next</span></span></button>
</div>
</div>
</div>
</div>"
`;
